### PR TITLE
Feature/devise basic old (rewrite)

### PR DESCRIPTION
**Pull request by ErenWeng**
# Devise (#12)

| Model |      User       |
| :---: | :-------------: |
|       | 使用原生 devise |

|  Model  | Organization |
| :-----: | :----------: |
| string  |     name     |

|   Model   | Organizations_user |
| :-------: | :-------------------: |
| references |         User          |
| references |     Organization      |

- [x] Gem 安裝

- [x] Create User model

- rails g devise User

- rails g devise View (for user)

- [x] create Organization model

- [x] 建立 Organization 與 user 關聯

- [x] Validation

- User model: organization presence/ email (uniqueness haven't in spec)
- Organization model: name

- [x] Spec

- User model
- Organization

- [x] fix model name
- [x] fix validation
- [x] fix spec

### CODE REVIEW

**Comment by koten0224** [comment]
一些該改的還是沒改唷～

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+  
+  #associations
+  has_many :user_organization_refs
+  has_many :organizations, through: :user_organization_refs
+
+  #validations
+  validates :email, presence: true, uniqueness: true, length: { minimum: 1, maximum: 100 }
+  validates :encrypted_password, presence: true, length: { minimum: 1, maximum: 100 }
```
Comment (RocketSH) [inline app/models/user.rb]: email 驗證regular expression? devise 是否會幫我們加密？

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -10,6 +10,16 @@
   </head>
 
   <body>
+    <nav>
+      <% if current_user %>
```
Comment (RocketSH) [inline app/views/layouts/application.html.erb:14]: Consider adding a helper method named `signed_in?` with like so:
```ruby
class ApplicationController
  # ...

  helper_method :current_user, :signed_in?

  def signed_in?
    !!current_user
  end
end
```

File: app/models/user.rb (hunk 1 above)
Comment (koten0224) [reply app/models/user.rb]: 會

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (ErenWeng) [reply app/views/layouts/application.html.erb:14]: 這個要看controller怎麼寫

Comment (koten0224) [reply app/views/layouts/application.html.erb:14]: 都可以，看用在什麼情境

**Pull request closed without merging**